cpmpy/solvers/exact.py

```diff
@@ -0,0 +1,528 @@
+#!/usr/bin/env python
+#-*- coding:utf-8 -*-
+##
+## exact.py
+##
+"""
+    Interface to Exact
+
+    Exact solves decision and optimization problems formulated as integer linear programs. Under the hood, it converts integer variables to binary (0-1) variables and applies highly efficient propagation routines and strong cutting-planes / pseudo-Boolean conflict analysis.
+
+    https://gitlab.com/JoD/exact
+
+    ===============
+    List of classes
+    ===============
+
+    .. autosummary::
+        :nosignatures:
+
+        CPM_exact
+"""
+import sys  # for stdout checking
+import time
+
+from .solver_interface import SolverInterface, SolverStatus, ExitStatus
+from ..expressions.core import *
+from ..expressions.variables import _BoolVarImpl, NegBoolView, _IntVarImpl, _NumVarImpl, intvar
+from ..transformations.comparison import only_numexpr_equality
+from ..transformations.flatten_model import flatten_constraint, flatten_objective, get_or_make_var
+from ..transformations.get_variables import get_variables
+from ..transformations.decompose_global import decompose_global
+from ..transformations.linearize import linearize_constraint, only_positive_bv
+from ..transformations.reification import only_bv_implies, reify_rewrite
+from ..expressions.globalconstraints import DirectConstraint
+import numpy as np
+import numbers
+
+class CPM_exact(SolverInterface):
+    """
+    Interface to the Python interface of Exact
+
+    Requires that the 'exact' python package is installed:
+    $ pip install exact
+    See https://pypi.org/project/exact for more information.
+
+    Creates the following attributes (see parent constructor for more):
+    - xct_solver: the Exact instance used in solve() and solveAll()
+    - assumption_dict: maps Exact variables to (Exact value, CPM assumption expression)
+    to recover which expressions were in the core
+    - solver_is_initialized: whether xct_solver is initialized
+    - self.objective_given: whether an objective function is given to xct_solver
+    - self.objective_minimize: the direction of the optimization (if false then maximize)
+    as Exact can only minimize
+    """
+
+    @staticmethod
+    def supported():
+        # try to import the package
+        try:
+            import exact
```

Comment:
Should we check for version here? 
People might have an older version of the Python API? (Which was the case for me ;) )

Reply:
done